NuRadioReco/modules/cwFilterNotch.py

```diff
@@ -0,0 +1,226 @@
+
+    return freq[peak_idxs]
+
+
+def filter_cws(trace : np.ndarray, freq : np.ndarray, spectrum : np.ndarray, fs = 3.2e9 * units.Hz, quality_factor = 1e3, threshold = 4):
+    """
+    Function that applies a notch filter at the frequency peaks of a given time trace
+    using the scipy library
+
+    Parameters
+    ----------
+    trace : np.ndarray
+        waveform (shape: [24,2048])
+    freq : np.ndarray
+        frequency of the trace's real fourier transform
+    spectrum:
+        the trace's real fourier transform
+    fs : float, default = 3.2e9 Hz
+        sampling frequency of the RNO-G DAQ
+    quality_factor : int, default = 1000
+        quality factor of the notch filter, defined as the ratio f0/bw, where f0 is the centre frequency
+        and bw the bandwidth of the filter at (f0,-3 dB)
+    threshold : int, default = 4
+        threshold for peak definition. A peak is defined as a point in the frequency spectrum
+        that exceeds threshold * rms(real fourier transform)
+
+    """
+    freqs = find_frequency_peaks(freq, spectrum, fs= fs, threshold=threshold)
+
+    if len(freqs) !=0:
+        notch_filters = [signal.iirnotch(freq, quality_factor, fs = fs) for freq in freqs]
+        trace_notched = signal.filtfilt(notch_filters[0][0], notch_filters[0][1], trace)
+        for notch in notch_filters[1:]:
+            trace_notched = signal.filtfilt(notch[0], notch[1], trace_notched)
+        return trace_notched
+
+    return trace
+
+
+def plot_trace(channel, ax, fs = 3.2e9 * units.Hz, label = None, plot_kwargs = dict()):
+    """
+    Function to plot trace of given channel
+    
+    Parameters
+    ----------
+    channel : NuRadio channel class
+        channel from which to get trace
+    ax : matplotlib.axes
+        ax on which to plot
+    fs : float, default = 3.2e9 Hz
+        sampling frequency of the RNO-G DAQ
+    label : string
+        plotlabel
+    plot_kwargs : dict
+        options for plotting
+    """
+    times = np.arange(2048)/fs / units.ns
+    trace = channel.get_trace()
+
+    legendloc = 2
+
+    ax.plot(times, trace, label = label, **plot_kwargs)
+    ax.set_xlabel("time / ns")
+    ax.set_ylabel("trace / V")
+    ax.legend(loc = legendloc)
+
+
+def plot_ft(channel, ax, label = None, plot_kwargs = dict()):
+    """
+    Function to plot real frequency spectrum of given channel
+    
+    Parameters
+    ----------
+    channel : NuRadio channel class
+        channel from which to get trace
+    ax : matplotlib.axes
+        ax on which to plot
+    label : string
+        plotlabel
+    plot_kwargs : dict
+        options for plotting
+    """
+    freqs = channel.get_frequencies()
+    spec = channel.get_frequency_spectrum()
+
+    legendloc = 2
+
+    ax.plot(freqs, np.abs(spec), label = label, **plot_kwargs)
+    ax.set_xlabel("freq / GHz")
+    ax.set_ylabel("amplitude / V/GHz")
+    ax.legend(loc = legendloc)
+
+
+class cwFilterNotch():
```

Comment:
typically we encode in the name on what the module acts on. So for example `channelCWNotchFilter`